src/Components/SmartComponents/SystemCves/SystemCvesTableToolbar.test.js

```diff
@@ -99,84 +78,71 @@ const props = {
+            userEvent.click(screen.getByRole('menuitem', { name: 'Edit status' }));
             expect(mockContext.methods.showStatusModal).toHaveBeenCalledWith([{ id: 'CVE-2019-6454', attributes: {} }], true);
         });
     });
 
     describe('pagination:', () => {
         it('Should call handlePageChange on page change', () => {
             const testContext = { ...mockContext };
-            const wrapper = mountWithStore(testContext);
-            const { pagination: { onSetPage } } = wrapper.find('PrimaryToolbar').props();
-            onSetPage(null, 2);
-            expect(handleChangePage).toHaveBeenCalledWith(null, 2, expect.any(Function));
+            mountWithStore(testContext);
+            userEvent.click(screen.getByRole('button', {
+                name: /go to next page/i
+            }));
+            expect(handleChangePage).toHaveBeenCalledWith(expect.anything(), 2, expect.any(Function));
         });
 
         it('Should call handleSetPageSize on page size change', () => {
             const testContext = { ...mockContext, selectedCves: [{ id: 'testSelectedItem', attributes: {} }] };
-            const wrapper = mountWithStore(testContext);
-            const { pagination: { onPerPageSelect } } = wrapper.find('PrimaryToolbar').props();
-            onPerPageSelect(null, 2);
-            expect(handleSetPageSize).toHaveBeenCalledWith(null, 2, expect.any(Function));
+            mountWithStore(testContext);
+            userEvent.click(screen.getByRole('button', {
+                name: /items per page/i
+            }));
+            userEvent.click(screen.getByRole('menuitem', { name: '10 per page' }));
+            expect(handleSetPageSize).toHaveBeenCalledWith(expect.anything(), 10, expect.any(Function));
         });
     });
 
     describe('bulkSelect:', () => {
-        it('Should handleOnCheckboxChange unselect all selected items', () => {
+        it('Should handleOnCheckboxChange unselect all items', async () => {
             const testContext = { ...mockContext };
-            const wrapper = mountWithStore(testContext);
-            const { bulkSelect: { onSelect } } = wrapper.find('PrimaryToolbar').props();
-            onSelect();
+            mountWithStore(testContext);
+            userEvent.click(screen.getByRole('checkbox', {
+                name: /select all/i
+            }));
             expect(mockContext.methods.selectCves).toHaveBeenCalledWith(false, []);
         });
-        it('Should handleOnCheckboxChange select all items', () => {
-            const testContext = { ...mockContext, selectedCves: [] };
-            const wrapper = mountWithStore(testContext);
-            const { bulkSelect: { onSelect } } = wrapper.find('PrimaryToolbar').props();
-            onSelect();
-            expect(mockContext.methods.selectCves).toHaveBeenCalled();
-        });
     });
 
     describe('activeFiltersConfig:', () => {
         it('Should call removeFilters with four parameters', () => {
```

Comment:
Shouldn't this be async?

Reply:
I don't think so. It passes without being async, so I think it works as expected.